agent/__init__.py

```diff
@@ -287,28 +288,40 @@ def send_ping(debug=False, dev=False):
     if not ping.ok:
         print('Ping failed.')
         return
-    blocklist = ping.json()
-    security_helper.block_ports(blocklist.get('block_ports', {'tcp': [], 'udp': []}))
-    security_helper.block_networks(blocklist.get('block_networks', []))
 
     connections, ports = security_helper.netstat_scan()
     payload = {
         'device_operating_system_version': platform.release(),
         'fqdn': socket.getfqdn(),
         'ipv4_address': get_primary_ip(),
         'uptime': get_uptime(),
-        'scan_info': ports,
-        'netstat': connections,
-        'processes': security_helper.process_scan(),
-        'firewall_enabled': security_helper.is_firewall_enabled(),
-        'firewall_rules': security_helper.get_firewall_rules(),
-        'selinux_status': security_helper.selinux_status(),
-        'app_armor_enabled': security_helper.is_app_armor_enabled(),
-        'logins': journal_helper.logins_last_hour(),
-        'default_password': security_helper.check_for_default_passwords(CONFIG_PATH),
         'agent_version': str(__version__)
     }
 
+    # Things we can't do within a Snap or Docker
+    if not (CONFINEMENT['snap'] or CONFINEMENT['docker']):
+        payload += {
+            'processes': security_helper.process_scan(),
+            'logins': journal_helper.logins_last_hour(),
+            'default_password': security_helper.check_for_default_passwords(
+                CONFIG_PATH),
+        }
+
+    # Things we cannot do in Docker
+    if not CONFINEMENT['docker']:
+        blocklist = ping.json()
+        security_helper.block_ports(blocklist.get('block_ports', {'tcp': [], 'udp': []}))
+        security_helper.block_networks(blocklist.get('block_networks', []))
+
+        payload += {
```

Comment:
`TypeError: unsupported operand type(s) for +=: 'dict' and 'dict'`

Reply:
Ok, this is just me being stupid. Of course.Fixing.